### PR TITLE
solution: 0015-3Sum-medium added

### PR DESCRIPTION
**Pull request by saloni33**
## Change Summary

Provide summary of changes with issue number if any.

## Checklist

**If you haven't fulfilled the below requirements, your PR won't be reviewed.**

### General 

- [x] Make sure you've read [CONTRIBUTING.md](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md) 
- [x] Make sure you've followed the template format
- [x] Make sure you've started the app locally and verified all the content and all links (if applicable) must be accessible correctly
- [x] Make sure you've applied LaTex for variables, formulas and time / space complexity instead of using backticks
- [x] Make sure you've formatted your code well (Code Block is rendered using `<Tabs/>`)
      Below is a C++ example. You can include multiple langs using `<TabItem/>` if necessary
      
````
<Tabs>

<TabItem value="cpp" label="C++">
<SolutionAuthor name="@YOUR_ALIAS"/>

```cpp
// Your code goes here
// Prefer K&R Coding style
```

</TabItem>

</Tabs>
````
- [x] Make sure you've explained your code well (write comments above each line - not inline)
- [x] Make sure you've written your explanation well and it is easy to understand for beginners
- [x] Make sure you've included Time & Space Complexity Analysis
- [x] Make sure you've included a meaningful approach name for your solution. Don't leave it blank. e.g. `## Approach 1: Two Pointers`.

### For Tutorial Articles

- [ ] Make sure you've explained your topic using 1 - 3 LC problems rather than using external problems. Solutions must be taken with explanations as well.
- [ ] Make sure you've provided suggested problems at the end using the given format. See [here](https://raw.githubusercontent.com/wingkwong/leetcode-the-hard-way/main/tutorials/math/number-theory/binary-exponentiation.md) as an example. If the target solution is not available, leave `solutionLink` blank.
- [ ] Make sure you've given credits / references if you use external resources. For an image, give credit under it. Otherwise, add a new section called References at the end (after Suggested Problems).

### CODE REVIEW

**Comment by saloni33** [comment]
@vigneshshiv kindly review

File: solutions/0000-0099/0015-3Sum.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+## Approach 1: Two Pointers
+ 
+We are required to find triplets which will have sum equal to zero.  
+Since the array contains positive and negative numbers both, first we will sort the array, by doing this our searching will become easy.  
+We need to find three numbers whose sum will be zero hence we will run a loop from $0$ to $n-2$ and fix a number each time, and find other two numbers after it.  
+If the fixed number is repeated again, we will skip the $for$ loop and move to next number.  
+Now, to find remaining two numbers, we will intialize two pointers $j$ and $k$ by $j = i + 1$ and $k = n - 1$ where $i$ is the index of fixed number and start our search similar to binary search method.  
+If $nums[i] + nums[j] + nums[k] == 0$ and this triplet is not present in our set (in order to remove duplicates) we will insert this triplet in our $ans$ vector.  
+If $nums[j] + nums[k] < target$, we need more positive numbers to make it 0, we will increament $j$ else we need more -negative numbers to make it 0, hence decreament $k$.
```
Comment (wingkwong) [inline solutions/0000-0099/0015-3Sum.md:57]: these will be packed together. Run the app locally to check. Don't use a markdown viewer.

File: solutions/0000-0099/0015-3Sum.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                        j++;
+                } else if (nums[j] + nums[k] < target) {
+                    j++;
+                } else {
+                    k--;
+                }
+            }
+        }
+        return ans;
+    }
```
Comment (wingkwong) [inline solutions/0000-0099/0015-3Sum.md:98]: 1. need full code. i.e. include Class Solution .. not just the main function
2. need comments on your code

**Review by wingkwong — CHANGES_REQUESTED** [review]

**Comment by saloni33** [comment]
Updated.

**Review by wingkwong — CHANGES_REQUESTED** [review]
1. your file name is incorrect. missing the difficulty at the end which causes the wrong ordering
2. put the time & space complexity before the code block
3. verify it locally first. it looks awful.
image

**Comment by wingkwong** [comment]
Is it ready for review?

**Comment by saloni33** [comment]
> Is it ready for review?

Yes I have required changes

**Comment by wingkwong** [comment]
Please click re-request review next time. Otherwise I'd assume it's still in progress.

**Comment by saloni33** [comment]
> Please click re-request review next time. Otherwise I'd assume it's still in progress.

okay

File: solutions/0000-0099/0015-3Sum-medium.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+vector<vector<int>> threeSum(vector<int>& nums) {
+        vector<vector<int>> ans;
+        set<vector<int>> st;
+        int n = nums.size();
+        
+        // sorted array
+        sort (nums.begin(), nums.end());
+        
+        // traversing the array
+        for (int i = 0 ; i < n-2 ; i++) {
```
Comment (wingkwong) [inline solutions/0000-0099/0015-3Sum-medium.md:78]: `n - 2`

File: solutions/0000-0099/0015-3Sum-medium.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+            
+            // skip the below code if i and i-1 element are same
+            if (i > 0 && nums[i] == nums[i-1])
```
Comment (wingkwong) [inline solutions/0000-0099/0015-3Sum-medium.md:81]: `i - 1`

**Review by wingkwong — CHANGES_REQUESTED** [review]

**Comment by saloni33** [comment]
@wingkwong kindly check

**Review by wingkwong — CHANGES_REQUESTED** [review]

**Pull request closed without merging**